python/ray/autoscaler/_private/aws/config.py

```diff
@@ -489,29 +501,70 @@ def _configure_subnet(config):
         ]
         if not subnets:
             cli_logger.abort(
-                "No usable subnets matching availability zone {} found.\n"
-                "Choose a different availability zone or try "
-                "manually creating an instance in your specified region "
-                "to populate the list of subnets and trying this again.",
-                config["provider"]["availability_zone"],
+                f"No usable subnets matching availability zone {azs} found "
+                f"for node type {node_type_key}.\nChoose a different "
+                "availability zone or try manually creating an instance in "
+                "your specified region to populate the list of subnets and "
+                "trying this again."
+            )
+        elif _are_user_subnets_pruned(subnets):
+            cli_logger.abort(
+                f"MISMATCH between specified subnets and Availability Zones! "
+                "The following Availability Zones were specified in the "
+                f"`provider section`: {azs}.\n The following subnets for node "
+                f"type `{node_type_key}` have no matching availability zone: "
+                f"{_get_pruned_subnets(subnets)}. "
```

Comment:
```suggestion
                f"{list(_get_pruned_subnets(subnets))}."
```
or, equivalently:
```python
                "{}.", _get_pruned_subnets(subnets)
```
`cli_logger` treats the default set-serialization format using curly brackets as a string format identifier otherwise, leading to an error of the following form:
```
  File "/Users/pdames/workplace/pdames/ray-project/ray/python/ray/autoscaler/_private/cli_logger.py", line 593, in abort
    self._error(msg, *args, _level_str="PANIC", **kwargs)
  File "/Users/pdames/workplace/pdames/ray-project/ray/python/ray/autoscaler/_private/cli_logger.py", line 558, in _error
    self.print(cf.red(msg), *args, _level_str=_level_str, **kwargs)
  File "/Users/pdames/workplace/pdames/ray-project/ray/python/ray/autoscaler/_private/cli_logger.py", line 579, in print
    self._print(_format_msg(msg, *args, **kwargs), _level_str=_level_str, end=end)
  File "/Users/pdames/workplace/pdames/ray-project/ray/python/ray/autoscaler/_private/cli_logger.py", line 230, in _format_msg
    return numbering_str + msg.format(*args, **kwargs) + tags_str
KeyError: "'subnet-000000'"
```